modules/azure/src/main/clojure/xtdb/azure/object_store.clj

```diff
@@ -24,7 +24,15 @@
      (catch BlobStorageException e
        (if (= 404 (.getStatusCode e))
          (throw (os/obj-missing-exception blob-name))
-         (throw e)))))
+         (throw e)))
+     (catch Throwable e
+       (let [unwrapped-e (Exceptions/unwrap e)]
```

Comment:
we have a macro, `util/rethrowing-cause` which does this unwrapping - if you wrap the code on L21-23 with that, you can then use a standard `catch` clause to check for `InterruptedException`

Reply:
Wrote up a new one in `azure.clj` - `rethrowing-reactor-cause` :)